### ISSUE TITLE
initial actions

### ISSUE THREAD

**Issue by leonwilly** [statelyai/xstate#466]
I just started using xstate today.  I must say it looks fantastic and documentation is excellent as well. That being said I quickly ran into an issue and after looking at the scxml documentation I believe there is a crucial missing feature. the "initial" property in the machine config should be either a string representation (short hand transition) or allowed to be a transition object.

**random example**
```js
const robots = []; // array of weak references
const genericRobot = Machine({
        id: "genericRobot",
        context: {
            name: 'generic'
        },
        initial: {target: "off", actions: 'addRobot'},
        states: {
            off : {}
        }
        //...
    },
    {
        actions: {
            addRobot(context) {
                robots.push(context.name)
            }
        }
    });


const sockeater = interpret(genericRobot.withContext({name: "sockeater"})).start();

```
Here is the excerpt from the W3C SCXML Specification.
 
> 3.6 <initial>
> [This section is normative.]
> 
> This element represents the default initial state for a complex <state> element (i.e. one containing child <state> or <parallel> elements.
> 
> 3.6.1 Attribute Details
> None
> 
> 3.6.2 Children
> <transition> A transition whose 'target' specifies the default initial state(s). Occurs once. In a conformant SCXML document, this transition must not contain 'cond' or 'event' attributes, and must specify a non-null 'target' whose value is a valid state specification consisting solely of descendants of the containing state (see 3.11 Legal State Configurations and Specifications for details). **This transition may contain executable content.**

_emphasis mine_

Maybe I missed this somewhere but just glancing at the source and trying it I don't think this is implemented.  The best part of the top initial transition is it can only be fired once. Technically this could be circumvented using a transient state but that seems more verbose and not as concise.

**Comment by gabewillen** [comment]
Any thoughts on this?  I looked at implementing this but the `initial` property is  deeply rooted and this requires a lot of changes to allow also using a `transition` object.  I still think its needed and should be included.  Right now I have to workaround this by using a transient `state` named initial in most of the machines I write.

**Comment by davidkpiano** [comment]
Definitely going to add this, the `initial` property can/should be a transition (non-guarded) with effects (executable content).

**Comment by gabewillen** [comment]
@davidkpiano Awesome thank you I will refactor my machines once this is resolved. There is no need to rush It does work properly with the transient state workaround.

**Comment by Andarist** [comment]
Just want to let you know that I've started looking into the source code to implement this.

**Comment by leonwilly** [comment]
Hey everyone i'm back to using this for another project. I saw this was closed however the initial specifier still doesn't accept a transaction object. Am I missing something?

**Comment by Andarist** [comment]
No, this wasn't implemented. I think we won't have the capacity to implement this in v4 but it's reasonable to implement for v5.

As a workaround, you can do smth like this:
```ts
createMachine({
    initial: 'first',
    states: {
        first: {
            always: {
                target: 'second',
                actions: 'doStuff'
            }
        },
        second: {}
    }
})
```
